[Devtools] Update Readme Links (#38)
devfile-samples/devfile-sample-go-basic: README.md

Before you begin creating an application with this `devfile` code sample, it's helpful to understand the relationship between the `devfile` and `Dockerfile` and how they contribute to your build. You can find these files at the following URLs:

> - [Sample Go `devfile.yaml`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/devfile.yaml)
> - [Sample Go `Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/docker/Dockerfile)
> - [Parent Go `devfile.yaml`](https://github.com/devfile/registry/blob/main/stacks/go/2.2.0/devfile.yaml)

Reviewer: ```suggestion
- [Go Sample `devfile.yaml`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/devfile.yaml)
- [Go Sample `Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/docker/Dockerfile)
- [Parent Go Stack `devfile.yaml`](https://github.com/devfile/registry/blob/main/stacks/go/2.2.0/devfile.yaml)
```

> This code sample has a parent devfile where it inherits most of the components and commands. The devfile contained in this repository is responsible for overwriting pieces of the parent devfile.

Reviewer: ```suggestion
This code sample inherits the components and commands of the specified parent devfile. The sample devfile overrides and extends parts of the parent devfile, in this case parts of the `deploy` component.
```

1. The parent `devfile.yaml` file has a [`build` component](https://github.com/devfile/registry/blob/main/stacks/go/2.2.0/devfile.yaml#L22-L28) that points to your `Dockerfile`.
2. The [`docker/Dockerfile`](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/docker/Dockerfile) contains the instructions you need to build the code sample as a container image.

> 3. The `devfile.yaml` [`deploy` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/devfile.yaml#L21-L33) overwrites the parent `deploy` component and points to a `deploy.yaml` file that contains instructions for deploying the built container image.

Reviewer: ```suggestion
3. The `devfile.yaml` [`deploy` component](https://github.com/devfile-samples/devfile-sample-go-basic/blob/v2.0.0/devfile.yaml#L21-L33) overrides the parent `deploy` component and points to a `deploy.yaml` file that contains instructions for deploying the built container image.
```
